pkg/controller/podautoscaler/horizontal.go

```diff
@@ -453,7 +453,12 @@ func (a *HorizontalController) reconcileAutoscaler(hpav1Shared *autoscalingv1.Ho
 		case desiredReplicas == 0:
 			//  never scale down to 0, reserved for disabling autoscaling
 			setCondition(hpa, autoscalingv2.ScalingLimited, v1.ConditionTrue, "TooFewReplicas", "the desired replica count was zero")
-			desiredReplicas = 1
+			// make sure we aren't below our minimum if one is defined
+			if hpa.Spec.MinReplicas != nil {
+				desiredReplicas = *hpa.Spec.MinReplicas
```

Comment:
Not sure, isn't this desired behaviour that low-usage defaults to scales to min replicas?

If I read https://kubernetes.io/docs/api-reference/v1.7/#horizontalpodautoscalerspec-v1-autoscaling correctly, `minReplicas` will default to `1` and can't be unset. Maybe the `desiredReplicas == 0` case could be completely dropped and just be ensured that `minReplicas` is set. In that case the `case hpa.Spec.MinReplicas != nil && desiredReplicas < *hpa.Spec.MinReplicas:` should catch also the `0` case and handle it correctly.